src/ld-query.js

```diff
@@ -47,13 +47,27 @@
 
         }, null );
 
+    function testObjectMatches( json, clause ) {
+
+        var val = json[ clause.where ];
+        if ( !val ) { return false; }
+        if ( isArray( val ) ) {
```

Comment:
I wonder if we should restrict this to just @type attributes? It doesn't really matter unless it helps with performance or clarity.